urllib3/util/selectors.py

```diff
@@ -534,20 +551,73 @@ def close(self):
             self._kqueue.close()
             super(KqueueSelector, self).close()
 
+    KqueueSelector = _KqueueSelector
+
+
+def _detect_select_patch(*args, **kwargs):
+    # If monkey-patching occurs then the canary won't be
+    # defined within the new select module. Because the
+    # module was patched we now have to re-evaluate which
+    # selectors are available.
```

Comment:
Ah! I was misinterpreting how eventlet works, looking at https://github.com/eventlet/eventlet/blob/master/eventlet/patcher.py#L290 confirms that you're correct. Perhaps I'll have to set this attribute on `select.select` instead of just `select`. If `select.select` isn't available then there's no selector anyway so no reason to check for patching.